### PR TITLE
Enable use of OpenTelemetry.Extensions.Hosting

### LINKED ISSUE #7007: Host Error with OpenTelemetry.Extensions.Hosting
When running `func start --csharp` when using `OpenTelemetry.Extensions.Hosting` in a function-app it gives the following error:

```bash
Microsoft (R) Build Engine version 16.7.0+7fb82e5b2 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  DoNothingApp -> /Users/kolanos/Web/newrelic/DoNothingApp/DoNothingApp/bin/output/DoNothingApp.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.69


Azure Functions Core Tools
Core Tools Version:       3.0.3160 Commit hash: 00aa7f43cc5c5f15241b5e6e5363256f19ceb990
Function Runtime Version: 3.0.14916.0

[2020-12-22T14:13:36.225Z] A host error has occurred during startup operation '8ad1dca7-4289-4911-ac9e-ac29c0d56a6f'.
[2020-12-22T14:13:36.225Z] func: Invalid host services. Microsoft.Azure.WebJobs.Script.WebHost: The following service registrations did not match the expected services:
[2020-12-22T14:13:36.225Z]   [Invalid] ServiceType: Microsoft.Extensions.Hosting.IHostedService, Lifetime: Singleton, ImplementationType: OpenTelemetry.Extensions.Hosting.Implementation.TelemetryHostedService, OpenTelemetry.Extensions.Hosting, Version=1.0.0.1, Culture=neutral, PublicKeyToken=7bd6737fe5b67e3c.
```

See [here](https://github.com/nintexvictor/DoNothingApp) for an example function app using `OpenTelemetry.Extensions.Hosting`.

cc @fabiocav

### PR DESCRIPTION
**Pull request by madelinegordon**
### Issue describing the changes in this PR

resolves #7007

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/WebJobs.Script.WebHost/DependencyInjection/DependencyValidator/DependencyValidator.cs
Comment (panuva-sympa) [inline src/WebJobs.Script.WebHost/DependencyInjection/DependencyValidator/DependencyValidator.cs]: The 3rd parameter is "assemblyPublicKeyToken" , which is wrong here. The correct one is **`"7bd6737fe5b67e3c"`** 
-- which you can get with "sn -T <assemblyName>" on VS developer command prompt, for example.

Please add a short comment here (something like enable OpenTelemetry.Net instrumentation library).

In addition, please add a row to the release_notes.md in the root of the repository like
- Enabled the use of OpenTelemetry.Net instrumentation library (#7253)

**Review by panuva-sympa — CHANGES_REQUESTED** [review]
This will be highly useful to get e.g. NewRelic and other instrumentations supported by OpenTelemetry.Net to work with Azure Functions. 

Please fix the assemblyPublicKeyToken to the correct one and add the row to release_notes.md. In addition please tick the boxes on the PR checklist in the PR description.

After these changes, let's try to get approval from the team.

File: src/WebJobs.Script.WebHost/DependencyInjection/DependencyValidator/DependencyValidator.cs
Comment (madelinegordon) [reply src/WebJobs.Script.WebHost/DependencyInjection/DependencyValidator/DependencyValidator.cs]: Thank you for your review!

**Review by fabiocav — DISMISSED** [review]
This looks good. Can you please go through the PR checklist above as well? Thanks!

**Review by ankitkumarr — DISMISSED** [review]

**Review by ankitkumarr — APPROVED** [review]

**Pull request merged**
